src/libmongoc/src/mongoc/mongoc-init.c

```diff
@@ -144,9 +146,11 @@ static BSON_ONCE_FUN (_mongoc_do_init)
 #endif
 
 #if defined(MONGOC_ENABLE_OCSP_OPENSSL)
-  _mongoc_ocsp_cache_init ();
+   _mongoc_ocsp_cache_init ();
 #endif
 
+   _mongoc_aws_credentials_cache_init ();
```

Comment:
Can/should this be guarded by a macro checking whether auth is necessary (i.e. `MONGOC_ENABLE_SSL`)?